src/components/List/List.js

```diff
@@ -0,0 +1,162 @@
+    const [filteredDataArray, setFilteredDataArray] = useState([]); //this array contains filtered group of data from the originalArray
+    const [totalCards, setTotalCards] = useState(0); //contains the total number of events' cards
+    const [cardsColorsArray, setCardsColorsArray] = useState([]);
+    const [change, setChange] = useState(false);
+    const navigate = useNavigate();
+
+
+    useEffect(() => {
+        loadEvents(); //Initially load all the events from the database
+    }, []);
+
+    useEffect(() => {
+        if (change) {
+            displayCards(); //load the events from the database to the originalDataArray
+            setChange(false);
+        }
+    });
+
+
+    //FUNCTIONS-----------------------------------------------------------------
+    const newButtonClickHandler = () => {
+        set(ref(db, 'eventToEdit/'), { //when user clicks new - no events to edit so make sure eventToEdit is set to "" empty string 
+            id: "",
+            name: "",
+            description: "",
+            date: ""
+        });
+        navigate('/form', { state: { id: 1, headerText: 'Create Event', buttonText: 'Add' } }); //navigate To Form Page
+    }
+
+    //This function loads all the cards/events and their count from the database to the page and checks for edits
+    const loadEvents = () => {
+        const dbRef = ref(getDatabase(app));
+        get(child(dbRef, 'events/')).then((snapshot) => {
+            if (snapshot.exists()) {
+                let originalData = [];
+                originalData = snapshotToArray(snapshot); //CONVERT FIREBASE DATABASE SNAPSHOT/COLLECTION TO AN ARRAY                displayCards(); //intially display the cards as the order they appear in the database
+                setOriginalDataArray(originalData);
+                setFilteredDataArray(originalData);
+                setTotalCards(originalData.length);
+                setChange(true);
+            } else {
+                console.log("No data available");
+            }
+        }).catch((error) => {
+            console.error(error);
+        });
+    }
+
+    //This function is used to display the data as cards on the screen
+    const displayCards = () => {
+        let colorsArray = [];
+        //set the colors array such that, Past date:red - Current date:blue - Future date:Purple  
+        for (let i = 0; i < filteredDataArray.length; i++) {
+            if (compareDates(filteredDataArray[i].date) === 1)
+                colorsArray[i] = "#982176"; //purple
+            else if (compareDates(filteredDataArray[i].date) === 0)
+                colorsArray[i] = "#1434A4"; //blue
+            else if (compareDates(filteredDataArray[i].date) === -1)
+                colorsArray[i] = "#D22B2B"; //red
+        }
+        setCardsColorsArray(colorsArray);
+    }
+
+    //This function is used when clicking a div to edit it's event
+    const divClickHandler = (event) => {
```

Comment:
it's better to rename to divClickHandler

Reply:
done